### DISCUSSION TITLE
How can I change the layout of plots in Jupyter?

### DISCUSSION THREAD

**Discussion by brycemagera** [quarto-dev/quarto-cli#6571] [Q&A]
### Description

The rest of my document is rendering great, but I have one cell block that I just can't get to render properly. It uses a for loop to render 6 different plots, and I would like to display them across two rows instead of having all of them in a single column. The only column layout that has changed anything at all is using `column: margin` - that works as expected.

I did see another post somewhere that stated using `plt.show()` instead of `df.show()` fixed the issue for them. Trying a similar fix here, unfortunately, did not work. Happy to provide any more details if needed.

```py
#| label: fig-sil-viz
#| fig-cap: "Silhouette score with varying number of clusters"
#| fig-subcap: true
#| column: screen-inset
#| layout-nrow: 2
#| width: 25%
cluster_list = range(2, 8)
for k in cluster_list:
    # finding optimal no. of clusters with silhouette coefficients
    visualizer = SilhouetteVisualizer(
        KMeans(k, random_state=1)
    )  ## Complete the code to visualize the silhouette scores for certain number of clusters
    visualizer.fit(k_means_df)
    # visualizer.show();
    plt.show()
```

Quarto version 1.3.450
Ubuntu 22.04.3 LTS

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by brycemagera** [reply]
Different dataset, but this should function the same way:

````qmd
---
title: Sample Quarto Document
format: html
cap-location: margin
---

This is a sample of a Quarto document producing plots from a for loop and rendering to HTML.

```{python}
#| label: fig-sil-viz
#| fig-cap: Silhouette score with varying number of clusters
#| fig-subcap: true
#| column: screen-inset
#| layout-nrow: 2
from sklearn.cluster import KMeans

from yellowbrick.cluster import silhouette_visualizer
from yellowbrick.datasets import load_credit

# Load a clustering dataset
X, y = load_credit()

# Specify rows to cluster: under 40 y/o and have either graduate or university education
X = X[(X['age'] <= 40) & (X['edu'].isin([1,2]))]

# Use the quick method and immediately show the figure
cluster_list = range(2, 8)
for k in cluster_list:
    v = silhouette_visualizer(KMeans(k, random_state=42), X, colors='yellowbrick')
    v.show()
```
END
````

**Reply by mcanouil** [reply]
The issue is because you want to have a nearly full page figures while still wanting caption in the margins.
if you change/remove `cap-location: margin` it works as expected.

@dragonstyle maybe there is something to do here as I think it should work.

**Reply by brycemagera** [reply]
Is there a way to make the figures smaller? I don't need them to span the whole page. I tried using `width: 50%` and other column options, but wasn't getting anything. Digging through the docs right now, I might find my answer there.

**Reply by mcanouil** [reply]
Don't set `column: screen-inset` if you don't need the figures to take the whole screen then.
But still, caption in a margin for a grid layout figures makes no sense.

**Reply by brycemagera** [reply]
The intent was for the caption of the whole grid to be in the margin, while the individual figures had the subcap underneath them. Quick diagram below. Note that it could be two columns or two rows, or whatever else, I just don't want a giant stack of plots.
I've tried every possible entry for column and the only one that produces a different result for me is `column: margin`, which does not give the desired output.

Fig 1    Fig 2
  (a)        (b)   
Fig 3    Fig 4        Figure 5: ...
  (c)        (d)
Fig 5    Fig 6
  (e)        (f)

**Reply by brycemagera** [reply]
I've discovered that adding `#| fig-cap-location: margin` to each individual figure and removing `cap-location` from the front matter will make this work as expected. What I don't understand is how that is any different from having `cap-location: margin` in the front matter.
![image](https://github.com/quarto-dev/quarto-cli/assets/52695653/2aaeb619-80a1-4f14-b71e-3c5c623b7993)

**Reply by mcanouil** [reply]
Indeed, this is why I asked Charles, since he is the one mostly behind the margin layout, as there might be a bug.

**Reply by dragonstyle** [reply]
The difference is indeed confusing. I will have a closer look
